tests/access/test_AccessControl.py

```diff
@@ -0,0 +1,176 @@
+import pytest
+from pathlib import Path
+from starkware.starknet.testing.starknet import Starknet
+from utils import (
+    Signer, TRUE, FALSE,
+    assert_event_emitted, assert_revert,
+    get_contract_def, cached_contract, contract_path
+)
+
+
+signer = Signer(123456789987654321)
+
+
+@pytest.fixture(scope='module')
+def contract_defs():
+    return {
+        Path(key).stem: get_contract_def(key)
+        for key in [
+            'openzeppelin/account/Account.cairo',
+            'tests/mocks/AccessControl.cairo',
+        ]
+    }
+
+
+@pytest.fixture(scope='module')
+async def accesscontrol_init(contract_defs):
+    starknet = await Starknet.empty()
+    account1 = await starknet.deploy(
+        contract_def=contract_defs['Account'],
+        constructor_calldata=[signer.public_key]
+    )
+    account2 = await starknet.deploy(
+        contract_def=contract_defs['Account'],
+        constructor_calldata=[signer.public_key]
+    )
+    accesscontrol = await starknet.deploy(
+        contract_def=contract_defs['AccessControl'],
+        constructor_calldata=[account1.contract_address]
+    )
+    return starknet.state, accesscontrol, account1, account2
+
+
+@pytest.fixture
+def accesscontrol_factory(contract_defs, accesscontrol_init):
+    state, accesscontrol, account1, account2 = accesscontrol_init
+    _state        = state.copy()
+    accesscontrol = cached_contract(_state, contract_defs['AccessControl'], accesscontrol)
+    account1      = cached_contract(_state, contract_defs['Account'], account1)
+    account2      = cached_contract(_state, contract_defs['Account'], account2)
+    return accesscontrol, account1, account2
+
+
+@pytest.mark.asyncio
+async def test_constructor(accesscontrol_factory):
+    accesscontrol, account1, account2 = accesscontrol_factory
+
+    expected1 = await accesscontrol.hasRole(0, account1.contract_address).call()
+    assert expected1.result.hasRole == TRUE
+
+    expected2 = await accesscontrol.hasRole(0, account2.contract_address).call()
```

Comment:
In solidity, we call it `DEFAULT_ADMIN_ROLE`. We can definitely reuse this name here, but I'm still not sure if the role should be `felt` or `uint256` in the context of cairo

Reply:
Please keep the role a `felt`. It will allow us to either use `grant_role('mint', caller)` or have the roles as consts or in a tidy namespace and use them from there (`grant_role(Roles.minter, caller)`). Dealing with `Uint256` is burdensome.